src/commands.rs

```diff
@@ -21,7 +21,8 @@ pub fn study(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult
     let ids = args
         .iter::<String>()
         .filter_map(Result::ok)
-        .filter_map(|x| roles.get_role_id(&x))
+        .map(|x| roles.get_role_id(&x.to_uppercase()))
```

Comment:
Regex can be case insensitive, using this allocates a new string for every role.
https://docs.rs/regex/1.2.1/regex/#grouping-and-flags

Reply:
This is for the individual courses, for instance, ``$study ea``. In this case, this string is directly used to search in the hashmap, where the keys all in uppercase. I guess I could use this just when this is the case